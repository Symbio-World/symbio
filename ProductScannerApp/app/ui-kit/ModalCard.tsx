import * as React from 'react'
import { View } from 'react-native'
import { t } from 'react-native-tailwindcss'

type ModalCardProps = {
  modalsInStack: number
}

export const ModalCard: React.FC<ModalCardProps> = ({
  children,
  modalsInStack,
}) => {
  return (
    <View
      style={[
        t.bgTransparent,
        t.wFull,
        t.hFull,
        t.itemsCenter,
        t.justifyCenter,
      ]}
    >
      <View
        style={[
          t.w10_12,
          t.bgWhite,
          t.roundedLg,
          { marginTop: 10 * modalsInStack },
        ]}
      >
        {children}
      </View>
    </View>
  )
}
