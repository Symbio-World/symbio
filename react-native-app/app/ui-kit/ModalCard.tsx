import * as React from 'react'
import { View } from 'react-native'
import { t } from 'react-native-tailwindcss'

type ModalCardProps = {}

export const ModalCard: React.FC<ModalCardProps> = ({ children }) => {
  return (
    <View
      style={[
        t.bgTransparent,
        t.wFull,
        t.hFull,
        t.itemsCenter,
        t.justifyCenter,
      ]}
    >
      <View
        style={[t.w11_12, t.bgWhite, t.roundedLg, t.shadowXl, t.overflowHidden]}
      >
        {children}
      </View>
    </View>
  )
}
