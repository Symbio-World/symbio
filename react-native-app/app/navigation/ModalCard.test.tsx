import * as React from 'react'
import { render } from 'react-native-testing-library'
import { Text } from 'react-native'

import { ModalCard } from './ModalCard'

describe('ModalCard', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <ModalCard>
        <Text>Test</Text>
      </ModalCard>,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
