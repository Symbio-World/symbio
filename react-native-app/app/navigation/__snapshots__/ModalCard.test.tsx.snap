// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCard renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0,0)",
      },
      Object {
        "width": "100%",
      },
      Object {
        "height": "100%",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "width": "91.666667%",
        },
        Object {
          "height": "95%",
        },
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "borderRadius": 8,
        },
        Object {
          "elevation": 12.5,
          "shadowColor": "rgba(0, 0, 0, 0.1)",
          "shadowOffset": Object {
            "height": 20,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 25,
        },
        Object {
          "overflow": "hidden",
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
</View>
`;
